bindings/rust/bench/src/harness.rs

```diff
@@ -1,163 +1,356 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{get_cert_path, PemType};
 use std::{
     cell::RefCell,
     collections::VecDeque,
     error::Error,
     fs::read_to_string,
     io::{ErrorKind, Read, Write},
-    rc::Rc,
+    rc::Rc, fmt::Debug,
 };
+use strum::EnumIter;
+
+#[derive(Clone, Copy, EnumIter)]
+pub enum PemType {
+    ServerKey,
+    ServerCertChain,
+    ClientKey,
+    ClientCertChain,
+    CACert,
+}
+
+impl PemType {
+    fn get_filename(&self) -> &str {
+        match self {
+            PemType::ServerKey => "server-key.pem",
+            PemType::ServerCertChain => "server-cert.pem",
+            PemType::ClientKey => "client-key.pem",
+            PemType::ClientCertChain => "client-cert.pem",
+            PemType::CACert => "ca-cert.pem",
+        }
+    }
+}
+
+#[derive(Clone, Copy, Default, EnumIter)]
+pub enum SigType {
+    Rsa2048,
+    Rsa3072,
+    Rsa4096,
+    #[default]
+    Ecdsa384,
+}
+
+impl SigType {
+    pub fn get_dir_name(&self) -> &str {
+        match self {
+            SigType::Rsa2048 => "rsa2048",
+            SigType::Rsa3072 => "rsa3072",
+            SigType::Rsa4096 => "rsa4096",
+            SigType::Ecdsa384 => "ecdsa384",
+        }
+    }
+}
+
+impl Debug for SigType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.get_dir_name())
+    }
+}
+
+pub fn get_cert_path(pem_type: PemType, sig_type: SigType) -> String {
+    format!(
+        "certs/{}/{}",
+        sig_type.get_dir_name(),
+        pem_type.get_filename()
+    )
+}
 
 pub fn read_to_bytes(pem_type: PemType, sig_type: SigType) -> Vec<u8> {
     read_to_string(get_cert_path(pem_type, sig_type))
         .unwrap()
         .into_bytes()
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[derive(Clone, Copy)]
 pub enum Mode {
     Client,
     Server,
 }
 
-#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+#[derive(Clone, Copy, Default, EnumIter, Eq, PartialEq)]
 pub enum HandshakeType {
     #[default]
     ServerAuth,
     MutualAuth,
 }
 
+impl Debug for HandshakeType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", match self {
+            Self::ServerAuth => "",
```

Comment:
It feels like there should be at least _something_ here. Perhaps just "server auth"

Reply:
Having the debug names be `ServerAuth` and `mTLS` or `ClientAuth` would make the bench group names like `handshake-ServerAuth-x25519-ecdsa384` and `handshake-ClientAuth-x25519-ecdsa384`. Ideally, I'd want the bench group names to be `handshake-x25519-ecdsa384` for non-mTLS and `handshake-mTLS-x25519-ecdsa384` for mTLS, but there's no good way to do that with debug names alone. (Using the backspace control character `\u{8}` as the debug string for `ServerAuth` did do the trick, but it was definitely too hacky.)

At first I was against it, but I think adding a little of matching logic to the handshake benchmark would make the most sense and be the most clear. I'll do that instead of manually implementing `Debug` for `HandshakeType`.